packages/x-tree-view/src/internals/plugins/useTreeViewItems/useTreeViewItems.tsx

```diff
@@ -126,6 +126,24 @@ export const useTreeViewItems: TreeViewPlugin<UseTreeViewItemsSignature> = ({
     (itemId: string) => selectorItemModel(store.value, itemId),
     [store],
   );
+  const toggleDisabledItem = useEventCallback((itemId: string, disabled: boolean) => {
+    if (itemId) {
```

Comment:
I think we should remove this check since `itemId` is not nullable.
Right now it should only skip when `item === ""` which strictly speaking is a valid id.